[AI-6169] DDS: Forescout Integration v1.0.0 (#21617)
DataDog/integrations-core: forescout/README.md

Integrate Forescout with Datadog's pre-built dashboard visualizations to gain insights into NAC Policy, Threat Protection, System Logs, and User Operations logs. With Datadog's built-in log pipeline, you can parse and enrich these logs to facilitate easy search and detailed insights. The integration can also be used for Cloud SIEM detection rules for enhanced monitoring and security.

**Minimum Agent version:** 7.73.0

## Setup

### Prerequisites
- The `Syslog` plugin must be installed in your Forescout setup.

### Configuration

#### Log Collection

1. Collecting logs is disabled by default in the Datadog Agent. Enable it in the `datadog.yaml` file with:

    ```yaml
      logs_enabled: true
    ```

2. Add this configuration block to your `forescout.d/conf.yaml` file to start collecting your Forescout logs:

   See the sample configuration file ([forescout.d/conf.yaml][8]) for available options.

    ```yaml
      logs:
        - type: tcp # or 'udp'
          port: <PORT>
          source: forescout
          service: forescout
    ```

> 

Reviewer: ```suggestion
    ```yaml
      logs:
        - type: tcp # or 'udp'
          port: <PORT>
          source: forescout
          service: forescout
    ```

   See the sample configuration file ([forescout.d/conf.yaml][8]) for available options.
```

Author: Done

**Note**:

> - `PORT`: Port should be similar to the port provided in `Configure syslog message forwarding from Forescout` section.

Reviewer: If the user has to complete this step first, I would move this reminder down to the lower section. Otherwise, it can be a bit confusing to say "this value should be similar to this other value ou haven't provided yet" 🙂

> - `PORT`: Port should be similar to the port provided in `Configure syslog message forwarding from Forescout` section.

Author: Aggred. We have moved down this note in the lower section and updated the content as required. Also, Datadog integration tile page does not support in-page link. Hence, we have not linked the section as recommended

> - Datadog recommends that you do not change the service and source values, as these parameters are integral to the pipeline's operation.

Author: Done

3. [Restart the Agent][3].

#### Configure timezone in Forescout

> **Note**: Follow this steps if Forescout appliance is not in the GMT timezone.

Reviewer: ```suggestion
Follow these steps if the Forescout appliance is not in the GMT timezone.
```

> 1. Login to the Forescout CounterACT appliance using root access via the command shell.

Reviewer: ```suggestion
1. Log in to the Forescout CounterACT appliance using root access via the command shell.
```

2. Run the timezone configuration command:
    ```sh
    fstool tz
    ```
3. Follow the setup prompts and configure the options as needed:
      ```text
      Select different time-zone? (yes/no): yes
      Choice (1-2) [1]: 2
      Enter the GMT offset (number between -14 and 12): 0
      Confirm Set time-zone to GMT? (yes/no) [yes]: yes

      Reboot is required to fully apply the time-zone change.
      Reboot now: yes
      ```

> 8. Wait until forescout reboot is completed.

Author: Done

#### Configure syslog message forwarding from Forescout

> 1. Log into the Forescout Console.

Reviewer: ```suggestion
1. Log in to the Forescout Console.
```

Author: Done

2. Navigate to **Tools > Options**.
3. Click **Syslog** from the options list.

> 4. Go to the **Send Events To** section and click Add.

Reviewer: ```suggestion
4. Go to the **Send Events To** section and click **Add**.
```

Author: Done

5. Enter the details as below for syslog configuration:
   - **Server Address**: Enter the IP address or hostname of the syslog server
   - **Server Port**: Specify the port number on which the syslog server is listening
   - **Server Protocol**: Select the protocol (UDP or TCP) to send the syslog messages
   - **Identity**: set this to `forescout-syslog`
   - **Facility**: set this to `syslog`

>    - **Severity**: set this to `info`

Author: Done

6. Click **OK**.
7. Go to the **Syslog Triggers** tab and configure the following:
   - Ensure the `Include only messages generated by the "Send Message to Syslog" action` checkbox is deselected.
   - Select `Short` in `Select format type for system log events and user operations` dropdown.

>    - Under `NAC Events`, `Threat Protection`, `System Logs and Events`, and `User Operations` include all the event types.

Author: Done
